### PR TITLE
[fix](auth)fix fe can not restart when replay create row policy log

### PR DESCRIPTION
**Pull request by zddr**
When executing two SQL statements simultaneously,such as
```
DROP ROW POLICY IF EXISTS test_row_policy_3 on zd.user1;CREATE ROW POLICY test_row_policy_3 ON zd.user1 AS RESTRICTIVE TO role role1 USING (k1 in (1) or k2 in (2));
```
fe will can not restart
```
Caused by: java.lang.ClassCastException: org.apache.doris.analysis.DropPolicyStmt cannot be cast to org.apache.doris.analysis.CreatePolicyStmt
	at org.apache.doris.policy.RowPolicy.gsonPostProcess(RowPolicy.java:169) ~[doris-fe.jar:1.2-SNAPSHOT]
	at org.apache.doris.persist.gson.GsonUtils$PostProcessTypeAdapterFactory$1.read(GsonUtils.java:640) ~[doris-fe.jar:1.2-SNAPSHOT]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:?]
	... 14 more
```
fix:
- setOrigStmt use split sql instead of full sql
- store index to cope with segmentation failure

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java [hunk 1]
```diff
@@ -102,6 +102,8 @@ public class RowPolicy extends Policy implements RowFilterPolicy {
      **/
     @SerializedName(value = "originStmt")
     private String originStmt;
+    @SerializedName(value = "stmtIdx")
+    private int stmtIdx;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java:106]: what will stmtIdx be when do deserialization if no stmtIdx in json

File: fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java [hunk 2]
```diff
@@ -218,7 +225,7 @@ public boolean isInvalid() {
     public Expression getFilterExpression() throws AnalysisException {
         NereidsParser nereidsParser = new NereidsParser();
         String sql = getOriginStmt();
-        CreatePolicyCommand command = (CreatePolicyCommand) nereidsParser.parseSingle(sql);
+        CreatePolicyCommand command = (CreatePolicyCommand) nereidsParser.parseMultiple(sql).get(stmtIdx).first;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java:228]: check out of bound error?

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java (hunk 1 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java:106]: will be default 0,we only support stmtIdx is 0

**Comment by zddr** [comment]
run performance

**Comment by zddr** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**